src/components/LoginApollo.js

```diff
@@ -42,47 +42,44 @@ const Login = ({ googleUser, setGoogleUser, clientID, location, history }) => {
   const handleLogin = googleUserLogin => {
     setGoogleUser(googleUserLogin);
     login()
-      .then()
-      .catch();
+      .then(res => {
+        const {
+          login: { token, login_status, register },
+        } = res.data;
+        if (login_status) {
+          if (sessionStorage.getItem("redirectTo").length > 0) {
+            window.location.replace(
+              process.env.NODE_ENV === "development"
+                ? `http://${sessionStorage.getItem("redirectTo")}?id=${token}`
+                : `https://${sessionStorage.getItem("redirectTo")}?id=${token}`
+            );
+          }
+        }
+        if (register) {
+          const user = readFromLocalStorage();
+          history.push({
+            pathname: "register",
+            state: {
+              user,
+              googleUser,
+              token: googleUser.getAuthResponse().id_token,
+            },
+          });
+        }
+      })
+      .catch(_ => {
+        alert("There was an error logging you in");
+      });
```

Comment:
I am more comfortable doing the login this way with regard to code readability.

Reply:
Let me know your opinions @98sean98

Reply:
from the standpoint of code execution flow, this makes more sense

also, regardless of using `useEffect()`, if `login()` takes longer than expected because of server latency, the app will appear frozen momentarily, we can still use `loading` from `useMutation` to return a loading animation right?